### PR TITLE
fix for Layout memory reporter uses pre-redirect url #6872

### LINKED ISSUE #6872: Layout memory reporter uses pre-redirect url
For example, if I look at the output for http://google.com the script reporter says "url(https://www.google.ca/?gfe_rd=cr&ei=kqK7Va3eD-eM8Qeyo4PwBQ)" while layout says "url(http://google.com/)".

### PR DESCRIPTION
**Pull request by amarant**
Hi, this fixes #6872
it has probably lot of problems, this is my first PR for servo, I'm waiting on your code review.
Thanks.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7612)

### CODE REVIEW

File: components/layout/layout_task.rs [hunk 1]
```diff
@@ -170,7 +170,7 @@ pub struct LayoutTask {
     pub id: PipelineId,
 
     /// The URL of the pipeline that we belong to.
-    pub url: Url,
+    pub url: RefCell<Option<Url>>,
```
Comment (eefriedman) [inline components/layout/layout_task.rs]: Why is this an `RefCell<Option<Url>>`, as opposed to just `RefCell<Url>`?  It looks like the code never stores None into the variable.

Comment (amarant) [inline components/layout/layout_task.rs]: Ok, I removed the Option as it was not needed.

**Comment by amarant** [comment]
So I removed the Option as it was not needed and make used of unwrap a lot.

**Comment by nox** [comment]
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

<sup>**[components/layout/layout_task.rs, line 539 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7612#-Jziqg4weKgpDMmT0RYJ)** ([raw file](https://github.com/servo/servo/blob/863f263c961529e7cf97a83518550700aba9f4e6/components/layout/layout_task.rs#L539)):</sup>
Don't let commented-out code in the tree.

---

<sup>**[components/layout/layout_task.rs, line 599 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7612#-JziqzILmhI951m0kN0T)** ([raw file](https://github.com/servo/servo/blob/863f263c961529e7cf97a83518550700aba9f4e6/components/layout/layout_task.rs#L599)):</sup>
You can just do:

```rust
*self.url.borrow_mut() = final_url;
```

In general, try to not name variables after their type, like `url_clone` or `url_ref_cell`.

---

<sup>**[components/script/layout_interface.rs, line 87 \[r2\]](https://reviewable.io:443/reviews/servo/servo/7612#-Jzis0Rt9lcVis-PazMg)** ([raw file](https://github.com/servo/servo/blob/f9412ca2d19a7e7101832ff328c6095c14705882/components/script/layout_interface.rs#L87)):</sup>
Is this always called only for the "final" URL? What happens when you navigate, following links? When the history API is used?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7612)

**Comment by nox** [comment]
@amarant Ping?

**Comment by KiChjang** [comment]
Rebased in #8903.

**Pull request closed without merging**